### DISCUSSION TITLE
Walk Drinking Custom Macro

### DISCUSSION THREAD

**Discussion by hossProjects** [HgAlexx/Buffet#27] [Q&A]
Hello,

I was wondering if it would be possible to filter the bests.conjuredDrink list to exclude Conjured Manna Biscuits if the player is moving.  I was toying around with a custom macro and the following:

```
if select(1, GetUnitSpeed("Player")) > 0 then
     bestdrink = "22018"
end
```

But it only seems to update/scan when the macro is pressed.  Would it be possible to check for player speed to filter the list?

**Comment by HgAlexx** [comment]
Hello,
The custom macro is run on specific event like when your bag content changes, when you change zone (to check for restricted items), etc.. and generate the macro(s). This macro update can only occur out of combat.
 
The generated macro itself is only executed when you click it.

I don't think you can do that within the macro because of the restriction imposed by Blizzard.

**Reply by hossProjects** [reply]
I get that the macro can only be use (and updated) out of combat.  Healers/Casters will from time to time use a method if drinking called "walk drinking" where they will sit long enough to get a tick and then move--this is all done out of combat.  What I am trying to figure out is what event handlers are triggering the scan to update the macro and if it would be possible to build a custom macro to filter the return values based on the GetUnitSpeed API.  

The use case here is:  If the player is moving to use conjured water first and then manna biscuits.

**Reply by HgAlexx** [reply]
In my knowledge, there is no trigger based on player movement.
For what you want, you need the speed test to be inside the macro, not to update the macro based on speed.
You can try with a custom macro, just make sure the speed test is part of the generated macro.
